src/gtk/toga_gtk/widgets/detailedlist.py

```diff
@@ -1,92 +1,185 @@
-from ..libs import Gtk
+from ..libs import Gtk, GLib, Gdk, Gio
 from .base import Widget
-from .internal.sourcetreemodel import SourceTreeModel
-from .internal.detailedlistrenderers import IconTextRenderer
+from .internal.rows import TextIconRow
+from .internal.buttons import RefreshButton, ScrollButton
 
 
+#Verify if right clicking a row currently works with touch screens, if not, use Gtk.GestureLongPress
 class DetailedList(Widget):
     """
-    gtk DetailedList implementation.
-    This is based on a Gtk.TreeView because the contents is still basic.
-    Should the need arise for more complex contents (buttons on each row, etc.),
-    we should use a Gtk.ListBox (see gtk3-demo).
+    Gtk DetailedList implementation.
+    Gtk.ListBox inside a Gtk.ScrolledWindow.
     """
 
     def create(self):
-        self.renderer = IconTextRenderer.from_style(self.interface)
+        self._on_refresh_handler = None
+        self._on_select_handler = None
+        self._on_delete_handler = None
```

Comment:
What is the purpose of these variables? If there's a user-provided handler, it should be declared and provide by the interface layer (i.e., `self.interface.on_refresh_handler`)

Reply:
It's the user handler that is set by `set_on_refresh`, `set_on_delete` and `set_on_select`. Those functions have a `handler: callable` argument.

Reply:
Sure - but why are they being copied here? Why not access them as self.interface.on_refresh_handler?